### ISSUE TITLE
Application start fails due to "java.lang.ClassCastException: io.dropwizard.util.Size cannot be cast to java.util.Optional"

### ISSUE THREAD

**Issue by maksymgendin** [dropwizard/dropwizard#1990]
Hi!

Has anyone seen this error already? I am using Dropwizard 1.1.0.
It happens while validating some fields (e.g. `inputBufferSize` field) in `io.dropwizard.jetty.HttpConnectorFactory` even if I do not specify a `server` block in my configuration YAML, so the default values are used.

```
java.lang.ClassCastException: io.dropwizard.util.Size cannot be cast to java.util.Optional
	at org.hibernate.validator.internal.engine.valuehandling.OptionalValueUnwrapper.handleValidatedValue(OptionalValueUnwrapper.java:20)
	at org.hibernate.validator.internal.engine.ValueContext.getCurrentValidatedValue(ValueContext.java:121)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:110)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:87)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.validateConstraint(MetaConstraint.java:73)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateMetaConstraint(ValidatorImpl.java:620)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraint(ValidatorImpl.java:583)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForSingleDefaultGroupElement(ValidatorImpl.java:527)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForDefaultGroup(ValidatorImpl.java:495)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForCurrentGroup(ValidatorImpl.java:460)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:410)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraint(ValidatorImpl.java:761)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraints(ValidatorImpl.java:684)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:419)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraint(ValidatorImpl.java:761)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraints(ValidatorImpl.java:684)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:419)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:207)
	at io.dropwizard.configuration.BaseConfigurationFactory.validate(BaseConfigurationFactory.java:237)
	at io.dropwizard.configuration.BaseConfigurationFactory.build(BaseConfigurationFactory.java:129)
	at io.dropwizard.configuration.BaseConfigurationFactory.build(BaseConfigurationFactory.java:92)
	at io.dropwizard.cli.ConfiguredCommand.parseConfiguration(ConfiguredCommand.java:124)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:72)
	at io.dropwizard.cli.Cli.run(Cli.java:74)
	at io.dropwizard.Application.run(Application.java:89)
```

The same also occurs sometimes for `io.dropwizard.util.Duration` (e.g. `idleTimeout` field).

Best regards
Maksym

**Comment by maksymgendin** [comment]
I think I found how you can reproduce this!
Enable TRACE level logging for `org.hibernate.validator`, so the line 108 in `org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree` gets executed and you get those ClassCastExceptions.

**Comment by maksymgendin** [comment]
For enabling TRACE in Hibernate you need `org.apache.logging.log4j:log4j-core` and `log4j2.xml` on your classpath.

`log4j2.xml` example:
```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="warn">
    <Appenders>
        <Console name="console" target="SYSTEM_OUT">
            <PatternLayout pattern="%d{yyy-MM-dd HH:mm:ss.SSS} [%t] %-5level %class{36}:%M(%L) - %msg%n"/>
        </Console>
    </Appenders>
    <Loggers>
        <Root level="trace">
            <AppenderRef ref="console"/>
        </Root>
    </Loggers>
</Configuration>
```

**Comment by joschi** [comment]
@maksymgendin Please provide a minimal project (Maven or Gradle build) to reproduce the problem.

**Comment by maksymgendin** [comment]
@joschi https://github.com/maksymgendin/dropwizard-error

**Comment by vyaasprashanth** [comment]
Even I'm getting the same error when I use log4j2

**Comment by tsheasha** [comment]
Hey, any idea when this will be released?

**Comment by arteam** [comment]
I think the issue is still open, the linked pull request doesn't fix it.

**Comment by tsheasha** [comment]
@joschi Any idea when a fix will be ready? Also any advice in the mean time as to how to work around this?

**Comment by nickbabcock** [comment]
I'll leave it to log4j2 users to post if they've found a workaround, but I have no idea when a fix will be ready as I don't know where and why this bug is occurring.

**Comment by maksymgendin** [comment]
The only workaround that I know is setting the log4j2 log level for Hibernate validator higher than TRACE.

**Comment by TaylorHG** [comment]
Is this still being tracked? I'd love to get my logs back.

**Comment by nickbabcock** [comment]
Investigated some more. This is not tied to log4j2, setting dropwizard-example test-example.yml to have
```yaml
logging:
    level: TRACE
```
will cause the failure. 

`DEBUG` will not cause the failure.

A failure only on a TRACE level is bizarre, I `git grep 'isTraceEnabled'` on hibernate validator (Dropwizard has no isTraceEnabled statements) and nothing stood out that it would involve a cast.

As a temporary fix, like @maksymgendin mentioned, one can set hibernate level to a level higher than `TRACE`

```yaml
logging:
    level: TRACE
    loggers:
       "org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree": DEBUG
```
This naturally leads me to suspect that issue is with the `isTraceEnabled` statement in `CosntraintTree` 🤔

**Comment by nickbabcock** [comment]
Reported upstream as [HV-1352](https://hibernate.atlassian.net/browse/HV-1352). Dropwizard users can use the provided workarounds until fixed upstream.

**Comment by joschi** [comment]
@maksymgendin I'm unable to reproduce the issue with Dropwizard 1.2.4 and 1.3.0-rc7, so I guess it has been solved by upgrading to a newer version of Hibernate Validator.

Please add a comment if the problem still occurs for you with the latest stable version of Dropwizard.

**Comment by maksymgendin** [comment]
@joschi Yeah, seems to be fixed in 5.4.2.Final. Thanks for reporting, @nickbabcock!
